### ISSUE TITLE
libretdec-libelf.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by prosteja** [avast-tl/retdec#100]
I install retdec manually in docker on clean ubuntu 17.10. following steps in Dockerfile(except I am using /app/retdec as installation prefix) Everything works fine unless I (re)move build directory. When I run retdec-decompiler.sh or just retdec-fileinfo after that I get the following error:
`/app/retdec/bin/retdec-fileinfo: error while loading shared libraries: libretdec-libelf.so: cannot open shared object file: No such file or directory`
This can help:
```
ldd /app/retdec/bin/retdec-fileinfo | grep retdec
	libretdec-libdwarf.so => /app/retdec/bin/../lib/libretdec-libdwarf.so (0x00007f1e0261d000)
	libretdec-libelf.so => not found
```
And:
```
readelf -d /app/retdec/bin/retdec-fileinfo | grep 'R.*PATH'
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib]
```
If I leave build dir in place:
```
ldd /app/retdec/bin/retdec-fileinfo | grep retdec
	libretdec-libdwarf.so => /app/retdec/bin/../lib/libretdec-libdwarf.so (0x00007f2695f7c000)
	libretdec-libelf.so => /retdec/build/external/src/libdwarf-project-install/lib/libretdec-libelf.so (0x00007f269508a000)
```
Thank you for your work!

**Comment by s3rvac** [comment]
Thank you for the report. I was able to reproduce it on Ubuntu 17.10 (Artful Aardvark) inside Docker.

The reason why it does not work is that CMake on Ubuntu 17.10 sets `RUNPATH` when installing RetDec while on other systems, it sets `RPATH`. The difference between them is that `RPATH` has a global effect, i.e. it is used to find both direct and transitive dependencies, while `RUNPATH` is used only to find direct dependencies. For more information, see e.g. [this post](https://stackoverflow.com/a/47243544/2580955).

We will have to either set a proper `RUNPATH` when installing `libretdec-dwarf.so` (currently, it points to the build directory), or force the use of `RPATH` instead of `RUNPATH` (I would not recommend this; rather, we should try to set a proper `RUNPATH` when installing `libretdec-dwarf.so`). We will let you know once this is fixed.

**Comment by s3rvac** [comment]
Fixed in b0a9829e8c53f138b9e70bf1bba2a81d71e3e879.

Details: The above commit updates the used version of our [libdwarf](https://github.com/avast-tl/libdwarf) repository. This new version contains commit [`85465d5`](https://github.com/avast-tl/libdwarf/commit/85465d5e235cc2d2f90d04016d6aca1a452d0e73), in which I have fixed the setting of runtime paths during installation. For `libdwarf`, we have to set the runtime path to `$ORIGIN` so it is able to find `libelf`, which is installed into the same directory as `libdwarf`.

I have verified the fix on Ubuntu 17.10 inside Docker. Current runtime paths:
```
$ readelf -d bin/retdec-fileinfo | grep 'R.*PATH'
0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../lib]
```
```
$ readelf -d lib/*.so | grep 'R.*PATH'
0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN]
0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN]
```
```
$ ldd bin/retdec-fileinfo | grep retdec
libretdec-libdwarf.so => /home/retdec/retdec-install/bin/../lib/libretdec-libdwarf.so (0x00007ffa7c80f000)
libretdec-libelf.so => /home/retdec/retdec-install/bin/../lib/libretdec-libelf.so (0x00007ffa7b91d000)
```
